plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -382,4 +360,45 @@ public void testEverythingImplemented()
             throw new AssertionError(e);
         }
     }
+
+    private static class CacheStatsAssertions
+    {
+        private final Supplier<CacheStats> stats;
+
+        private int loads;
+        private int hits;
+        private int misses;
+
+        public CacheStatsAssertions(Supplier<CacheStats> stats)
+        {
+            this.stats = requireNonNull(stats, "stats is null");
+        }
+
+        public CacheStatsAssertions loadsIncreasedBy(int value)
+        {
+            this.loads += value;
+            assertThat(stats.get().loadCount())
+                    .withFailMessage("Expected load count is %d but actual is %d", loads, stats.get().loadCount())
+                    .isEqualTo(loads);
+            return this;
+        }
+
+        public CacheStatsAssertions hitsIncreasedBy(int value)
+        {
+            this.hits += value;
+            assertThat(stats.get().hitCount())
+                    .withFailMessage("Expected hit count is %d but actual is %d", hits, stats.get().hitCount())
+                    .isEqualTo(hits);
+            return this;
+        }
+
+        public CacheStatsAssertions missesIncreasedBy(int value)
+        {
+            this.misses += value;
+            assertThat(stats.get().missCount())
+                    .withFailMessage("Expected miss count is %d but actual is %d", misses, stats.get().missCount())
+                    .isEqualTo(misses);
```

Comment:
We should also verify other stats. If you verify that `missesIncreasedBy` then you need to verify that load and hits didn't change. 

Maybe you need `loadsAndMissesIncreasedBy(2)`?

Reply:
PTAL now

Reply:
@kokosing, I think that it is just the duplication of logic. I'd get rid of that `verify()`

Reply:
We always need to verify all three metrics (or two, to me misses=loaded), however I don't like this additional call.

> (or two, to me misses=loaded)

If we would remove `loaded` metric verification, then you can inline `verify()` into `missesIncreasedBy()` and `hitsIncreasedBy()`

Reply:
I'm fine with a little duplicatiob